docs/blueprints/EP028.rst

```diff
@@ -0,0 +1,153 @@
+    |                     |        |                     |
+    |                     |        |                     |
+    +-----------^---------+        +---------------------+
+                |
+                | send
+    +-----------+---------+
+    |                     |
+    |                     |
+    | Filebeat service    |
+    |                     |
+    |                     |
+    +----------^----------+
+               |
+               | input source: log file
+               |
+    +----------+----------+
+    |                     |
+    |                     |
+    | kytos.log (file)    |
+    |                     |
+    |                     |
+    +----------^----------+
+               |
+               | write
+               |
+    +----------+----------+
+    |                     |
+    |                     |
+    |                     |
+    | RotatingFileHandler |
+    |                     |
+    | logging.ini         |
+    |                     |
+    |                     |
+    |                     |
+    |                     |
+    +---------------------+
+
+
+As represented in this diagram, as long as ``logging.ini`` has a handler that's writing to a file, Filebeat can read from it as an input, so it's up to network operators to configure it accordingly and also make sure to use the same formatter and dissect processors to parse and structure the fields of the log messages. For instance, this ``logging.ini`` below is based on `amlight/kytos-docker logging.ini <https://github.com/amlight/kytos-docker/blob/master/logging.ini>`_:
+
+.. code-block:: none
+
+    [formatters]
+    keys: file
+
+    [loggers]
+    keys: root
+
+    [formatter_file]
+    format: %(asctime)s - %(levelname)s [%(name)s] [%(filename)s:%(lineno)d:%(funcName)s] (%(threadName)s) %(message)s
+
+    [handler_file]
+    class: handlers.RotatingFileHandler
+    args:["/var/log/kytos/kytos.log", "a", 10*1024*1024, 5]
+    formatter: file
+    level: INFO
+
+    [logger_root]
+    level: INFO
+    handlers: file
+    propagate: True
+
+The following ``filebeat.yml`` configuration would correctly process, ingest and process the structured logs based on the expected placeholders of the file formatter, notice that the tokenizer with this configuration ``'%{kytos.timestamp} - %{kytos.level} [%{kytos.logger}] [%{kytos.file}:%{kytos.line}:%{kytos.func}] (%{kytos.thread}) %{kytos.message}'`` matches accordingly the expected fields in the formatter:
+
+.. code-block:: none
+
+    ---
+    filebeat.modules:
+    - module: system
+      syslog:
+        enabled: false
+      auth:
+        enabled: false
+
+    processors:
+      - dissect:
+          # kytos log formatter tokenizer to be able to search for fields easily
+          tokenizer: '%{kytos.timestamp} - %{kytos.level} [%{kytos.logger}] [%{kytos.file}:%{kytos.line}:%{kytos.func}] (%{kytos.thread}) %{kytos.message}'
```

Comment:
We need to align the formatter that we consider that has all the structured files that we consider essential in prod, this is a first draft based on what I've ended when developing locally and exploring. The difference compared to the logging.ini that it's being used on Dockerfile-prod is this extra `[%{kytos.file}:%{kytos.line}:%{kytos.func}]`, that will extract the file name, line number and function, I think when debugging hard problems this might become handy, let me know what you think and if you need other ones in prod that haven't been covered yet.

Reply:
Example of the additional information that were introduced with some captured logs:

```
2022-04-27 17:38:19,489 - INFO [kytos.core.api_server] [api_server.py:400:_start_endpoint] (MainThread) Started /api/amlight/flow_stats/bytes_count/<flow_id> - GET
2022-04-27 17:38:19,490 - INFO [kytos.core.api_server] [api_server.py:400:_start_endpoint] (MainThread) Started /api/amlight/flow_stats/bytes_count/per_flow/<dpid> - GET
2022-04-27 17:38:19,490 - INFO [kytos.core.api_server] [api_server.py:400:_start_endpoint] (MainThread) Started /api/amlight/flow_stats/bytes_count/sum/<dpid> - GET
2022-04-27 17:38:19,490 - INFO [kytos.core.api_server] [api_server.py:400:_start_endpoint] (MainThread) Started /api/amlight/flow_stats/flow/match/<dpid> - GET
2022-04-27 17:38:19,490 - INFO [kytos.core.api_server] [api_server.py:400:_start_endpoint] (MainThread) Started /api/amlight/flow_stats/flow/stats/<dpid> - GET
2022-04-27 17:38:19,490 - INFO [kytos.core.api_server] [api_server.py:400:_start_endpoint] (MainThread) Started /api/amlight/flow_stats/packet_count/<flow_id> - GET
2022-04-27 17:38:19,491 - INFO [kytos.core.api_server] [api_server.py:400:_start_endpoint] (MainThread) Started /api/amlight/flow_stats/packet_count/per_flow/<dpid> - GET
2022-04-27 17:38:19,491 - INFO [kytos.core.api_server] [api_server.py:400:_start_endpoint] (MainThread) Started /api/amlight/flow_stats/packet_count/sum/<dpid> - GET
2022-04-27 17:38:19,509 - INFO [kytos.core.controller] [controller.py:542:raw_event_handler] (MainThread) Raw Event Handler started
2022-04-27 17:38:19,510 - INFO [kytos.core.controller] [controller.py:558:msg_in_event_handler] (MainThread) Message In Event Handler started
2022-04-27 17:38:19,510 - INFO [kytos.core.controller] [controller.py:586:msg_out_event_handler] (MainThread) Message Out Event Handler started
2022-04-27 17:38:19,510 - INFO [kytos.core.controller] [controller.py:624:app_event_handler] (MainThread) App Event Handler started
```

Reply:
Sounds good. Indeed. @italovalcy I've mapped this point on this issue here https://github.com/amlight/kytos-docker/issues/15